go.mod

```diff
@@ -870,3 +870,5 @@ retract v0.37.0 // Contains dependencies on v0.36.0 components, which should hav
 
 // see https://github.com/distribution/distribution/issues/3590
 exclude github.com/docker/distribution v2.8.0+incompatible
+
+replace github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.12.2-0.20220318110013-3bc8f2c651ff
```

Comment:
Why do we need this? Is this needed on every module that imports the prometheus exporter module?

Reply:
not every module but some tests like `cmd/otelcontribcol/main_testbed.go` fails (because it runs prometheusexporter) and there is no `go.mod` in that directory to include it in there so I put it in the main go.mod

Reply:
🤔 I would expect that removing the `replace` and running `make gotidy` would fix this.

Reply:
I will have a look in the next few days to see if I can make it work. Using a replace probably means that downstream distributions will need to use this too, so if we can avoid it it would be great

Reply:
Hey, sorry for the delay. I have tried using the latest commit, but it seems like Go considers v1.12.2 to be higher than the latest commit and decides to use this. I am happy to approve this once 
1. the test suite passes and
2. you document why the `replace` is needed, so that downstream distros can understand what they need to do to accomodate this.

Reply:
Rebased to the latest main. Added a comment in go.mod explaining why `replace` is needed, please let me know if that's all that is needed there. Fixed couple more CI failures. Could you please rerun the CI?